common/SC_PaUtils.hpp

```diff
@@ -0,0 +1,22 @@
+// helper functions for SuperCollider server's PortAudio backend
+
+#pragma once
+
+#include "portaudio.h"
+#include <string>
+
+enum class IOType { Input, Output };
```

Comment:
perhaps, since this is going into the global namespace now, it makes more sense to remove this enum here and just pass a boolean `isInput` to `GetPaDeviceFromName`. but, you could still use this enum in the cpp file if you don't want to move too much code around -- just do `auto ioType = isInput ? IOType::Input : IOType::Output;` at the start of the function definition

Reply:
changing everything to `bool isInput` was easy enough so I did that